src/imitation/algorithms/adversarial.py

```diff
@@ -251,28 +250,31 @@ def _build_disc_feed_dict(self, *,
       tf.logging.debug("_build_disc_feed_dict: No generator rollout "
                        "parameters were "
                        "provided, so we are generating them now.")
-      gen_old_obs, gen_act, gen_new_obs = self._gen_replay_buffer.sample(
+      gen_sample = self._gen_replay_buffer.sample(
           self._n_disc_samples_per_buffer)
+      gen_old_obs = gen_sample.old_obs
+      gen_act = gen_sample.act
+      gen_new_obs = gen_sample.new_obs
```

Comment:
May as well delete these and `s/gen_\(old_obs\|act\|new_obs\)/gen_sample.\1/g`.

Reply:
I don't think we can do this as `gen_*`  can also be supplied as keyword arguments. I could change the argument to optionally take a `Transition` object instead of `gen_*`, but that would then force the user to provide some irrelevant attributes (e.g. `rew`, `done`).

Reply:
Oh, my suggestion was more minor than that: I don't mean passing `gen_sample` to downstream functions, I just mean referring to the attributes directly within this function instead of creating variables.

Reply:
This function, `_build_disc_feed_dict`, takes `gen_*` as arguments -- it only computes `gen_sample` if they're `None`. (This isn't obvious from the context provided in the diff.)

Reply:
Thank you for the clarification! I should have read more carefully.